### PR TITLE
C2 omar hossam c5 s17 admin create/remove categories

### PR DESCRIPTION
**Pull request by omar-mohyeldin**
issue #368 
@karimngr 
@nohahesham 
@Mirna-Yacout 
@AmrARaouf 
Please review

### CODE REVIEW

File: arability/app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -261,4 +261,73 @@ def upload
...
+
+  # Author:
+  #   Omar Hossam
+  # Description:
+  #   As an admin, I could delete any category from database by choosing it from
+  #   the view table and clicking the trash icon
+  # Parameters:
+  #   category_id: id of category to be deleted
+  # Success:
+  #   Category is deleted from database and categpries view appears without the
```
Comment (AmrARaouf) [inline arability/app/controllers/admin_controller.rb]: fix typo please

File: arability/config/routes.rb [hunk 2]
```diff
@@ -15,6 +15,10 @@
     post "/add_word"
     post "/add_trophy"
     post "/add_prize"
+
```
Comment (AmrARaouf) [inline arability/config/routes.rb:18]: delete space or merge the 2 blocks if they both belong to admin

Comment (omar-mohyeldin) [inline arability/config/routes.rb:18]: there are no spaces in this empty line, and this empty line is needed to differentiate between get, post and match routes, and to make it more organized, even though they all belong to the admin scope

**Comment by omar-mohyeldin** [comment]
@AmrARaouf please review again

**Comment by Mirna-Yacout** [comment]
@Omar-Hossam 
- retrieving categories should be requested by the controller from the model then returned (dashed line)
- returning arrow should be dashed (@category)

**Comment by AmrARaouf** [comment]
Doc reviewed

**Comment by omar-mohyeldin** [comment]
@Mirna-Yacout please re-review

**Comment by Mirna-Yacout** [comment]
- title
- @categories is also requested by controller from the model, returned dashed to controller

File: arability/app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -261,4 +261,73 @@ def upload
...
+  #   None
+  def delete_category
+    category_id = params[:category_id]
+    category = Category.find(category_id)
+    category.delete
+    flash[:success] = "لقد تم مسح الفئة بنجاح"
+    flash.keep
+    redirect_to action: 'all_category'
+  end
+
```
Comment (nohahesham) [inline arability/app/controllers/admin_controller.rb:400]: remove empty line

Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb:400]: this empty line is needed to differentiate between 2 different block of codes, and is needed in git merging

**Comment by omar-mohyeldin** [comment]
@nohahesham please review again

**Comment by omar-mohyeldin** [comment]
@Mirna-Yacout what do you mean by title? we never did use titles in our UMLs. So what do you mean by them and what is its content? And are we obliged to add them even though we never did in our previous UMLs?

**Comment by Mirna-Yacout** [comment]
by title I mean a title for your story to indicate which story u are implementing, as far as I recently knew (that's why I did not require it before), it's necassary

**Comment by AmrARaouf** [comment]
@Omar-Hossam please add your test files

**Comment by omar-mohyeldin** [comment]
@Mirna-Yacout please re-review UML
@AmrARaouf please review the tests

**Comment by omar-mohyeldin** [comment]
@nohahesham please re-review
@karimngr please review

**Comment by Mirna-Yacout** [comment]
verified

**Comment by nohahesham** [comment]
@Omar-Hossam  code verified

**Comment by ghost** [comment]
verified

**Comment by AmrARaouf** [comment]
@Omar-Hossam please resolve conflicts before I can merge your pull request

**Comment by omar-mohyeldin** [comment]
@AmrARaouf you could now merge

**Pull request merged**